tls/s2n_server_cert.c

```diff
@@ -45,8 +45,8 @@ int s2n_server_cert_recv(struct s2n_connection *conn)
     cert_chain.data = s2n_stuffer_raw_read(&conn->handshake.io, size_of_all_certificates);
     POSIX_ENSURE_REF(cert_chain.data);
 
-    POSIX_GUARD(s2n_x509_validator_validate_cert_chain(&conn->x509_validator, conn, cert_chain.data,
-                         cert_chain.size, &actual_cert_pkey_type, &public_key));
+    POSIX_ENSURE(s2n_x509_validator_validate_cert_chain(&conn->x509_validator, conn, cert_chain.data,
+                         cert_chain.size, &actual_cert_pkey_type, &public_key) == S2N_CERT_OK, S2N_ERR_CERT_UNTRUSTED);
```

Comment:
Why not use `S2N_ERR_CERT_NOT_VALIDATED` here?

Reply:
S2N_ERR_CERT_NOT_VALIDATED is a usage error implying you're trying to use an certificate that doesn't exist / hasn't been validated yet. S2N_ERR_CERT_UNTRUSTED is a protocol error indicating that we did validate the certificate, and that validation failed.